moveit_ros/moveit_servo/src/utils/common.cpp

```diff
@@ -0,0 +1,333 @@
+/*******************************************************************************
+ * BSD 3-Clause License
+ *
+ * Copyright (c) 2019, Los Alamos National Security, LLC
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *******************************************************************************/
+
+/*      Title     : utils.cpp
+ *      Project   : moveit_servo
+ *      Created   : 05/17/2023
+ *      Author    : Andy Zelenak, V Mohammed Ibrahim
+ */
+
+#include <moveit_servo/utils/common.hpp>
+
+namespace
+{
+// The threshold above which `override_velocity_scaling_factor` will be used instead of computing the scaling from joint bounds.
+const double SCALING_OVERRIDE_THRESHOLD = 0.01;
+}  // namespace
+
+namespace moveit_servo
+{
+
+bool isValidCommand(const Eigen::VectorXd& command)
+{
+  // returns true only if there are no nan values.
+  return (command.array() == command.array()).all();
+}
+
+bool isValidCommand(const Eigen::Isometry3d& command)
+{
+  bool is_valid_rotation = true;
+  Eigen::Matrix3d identity, rotation;
+  identity.setIdentity();
+  rotation = command.linear();
+  // checks rotation, will fail if there is nan
+  is_valid_rotation = identity.isApprox(rotation.inverse() * rotation);
+  // Command is not vald if there is Nan
+  const bool not_nan = isValidCommand(command.translation());
+  return is_valid_rotation && not_nan;
```

Comment:
assuming you can't just use `hasNaN()` to do everything for you in this function, at least change `not_nan` to `is_valid_translation` to be consistent. But definitely try the first.